utils.py

```diff
@@ -475,6 +477,45 @@ def get_human_readable_time_string(time_msec):
         '%B %d %H:%M:%S', time.gmtime(python_utils.divide(time_msec, 1000.0)))
 
 
+def create_string_from_largest_unit_in_timedelta(timedelta_obj):
+    """Given the timedelta object, find the largest non zero time unit and
+    return that value, along with the time unit, as a human readable string.
+    The returned string is not localized.
+
+    Args:
+        timedelta_obj: datetime.timedelta. A datetime timedelta object. Datetime
+            timedelta objects are created when you subtract two datetime
+            objects.
+
+    Returns:
+        str. A human readable string representing the value of largest non zero
+        time unit, along with the time units. If the largest time unit is
+        seconds, 1 minute is returned. The value is represented as an integer in
+        the string.
+
+    Raises:
+        Exception. If the provided timedelta is not positive.
+    """
+    total_seconds = timedelta_obj.total_seconds()
+    if total_seconds <= 0:
+        raise Exception(
+            'Expected a positive timedelta, received: %s.' % total_seconds)
+    elif timedelta_obj.days != 0:
+        return '%s day%s' % (
+            int(timedelta_obj.days), 's' if timedelta_obj.days > 1 else '')
+    else:
```

Comment:
Do we need `else` here?

Reply:
No but in the past Sean has asked me to keep the else there so I just put it there for readability